src/groupByKey.ts

```diff
@@ -2,6 +2,14 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey(items: [], key: string): GroupsMap<{}> {
+  const result: GroupsMap<{}> = {};
+
+  for (let i = 0; i < items.length; i += 1) {
+    const sort = items.filter((obj) => obj[key] === items[i][key]);
```

Comment:
'sort' is a reserved word like a map, var, or function 
so, make sense to rename it